crates/storage/provider/src/providers/snapshot/writer.rs

```diff
@@ -0,0 +1,227 @@
+                )
+            }
+            Err(err) => return Err(err),
+        };
+        Ok((NippyJarWriter::from_owned(jar)?, path))
+    }
+
+    /// Commits configuration changes to disk, and updates the filename to reflect the updated block
+    /// and/or tx ranges.
+    pub fn commit(&mut self) -> ProviderResult<()> {
+        let segment = self.writer.user_header().segment();
+        let block = self.writer.user_header().block_end();
+
+        // Commits offsets and new user_header to disk
+        self.writer.commit()?;
+
+        // Ensures block range and transaction range are flushed to the filename
+        let previous_path = self.data_path.clone();
+        let new_path = self
+            .reader
+            .directory()
+            .join(segment.filename_from_header(self.writer.user_header().clone()));
+
+        NippyJar::<SegmentHeader>::load(&previous_path)?.rename(new_path)?;
+
+        // Opens the updated snapshot
+        let (writer, data_path) = Self::open(segment, block, self.reader.clone())?;
+        self.writer = writer;
+        self.data_path = data_path;
+
+        Ok(())
+    }
+
+    /// Appends a column to disk. It creates and loads a new snapshot file if block goes beyond the
+    /// current block range.
+    ///
+    /// ATTENTION: It **requires** `self.commit` to be called manually
+    fn append<T, F1>(
+        &mut self,
+        block: BlockNumber,
+        segment: SnapshotSegment,
+        column: T,
+        initialize_segment: F1,
+    ) -> ProviderResult<()>
+    where
+        T: Compact,
+        F1: Fn(&mut SegmentHeader),
+    {
+        // We have finished the previous snapshot and must freeze it
+        if block > self.writer.user_header().block_end() {
+            // Commits offsets and new user_header to disk
+            self.writer.commit()?;
+
+            // We need to commit the changes done to `SegmentHeader` as well
+            let previous_path = self.data_path.clone();
+            let new_path = self
+                .reader
+                .directory()
+                .join(segment.filename_from_header(self.writer.user_header().clone()));
+
+            // Opens the new snapshot
+            let (writer, data_path) = Self::open(segment, block, self.reader.clone())?;
+            self.writer = writer;
+            self.data_path = data_path;
+
+            // Initializes the new segment header
+            initialize_segment(self.writer.user_header_mut());
+
+            // Updates the name of the previous snapshot
+            NippyJar::<SegmentHeader>::load(&previous_path)?.rename(new_path)?;
+        } else {
+            self.writer.user_header_mut().increment()
+        }
+
+        self.append_column(column)?;
+
+        Ok(())
+    }
+
+    /// Truncates a number of rows from disk. It delets and loads an older snapshot file if block
+    /// goes beyond the start of the current block range.
+    ///
+    /// ATTENTION: It commits in the end.
```

Comment:
why we don't commit on append, but commit on truncate?

Reply:
we do now